### PR TITLE
constexpr ldexp

### PR DESCRIPTION
**Pull request by mborland**
As requested by @ckormanyos

### CODE REVIEW

**Comment by ckormanyos** [comment]
> As requested by @ckormanyos

Thank you @mborland. This is very useful. It is amazing, how quickly and reliably you're implementing, testing and documenting these functions

**Comment by mborland** [comment]
@NAThompson and @jzmaddock Pending review this should be good. The two CI failures are network errors.

File: include/boost/math/ccmath/ldexp.hpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    {
+        if(result < denorm_min_limit)
+        {
+            throw std::underflow_error("Exponent leads to a value that is smaller than minimum positive subnormal value");
+        }
+
+        result /= 2;
+        --abs_exp;
+    }
+
```
Comment (NAThompson) [inline include/boost/math/ccmath/ldexp.hpp:42]: This whole loop is `constexpr`!?

File: include/boost/math/ccmath/ldexp.hpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+template <typename Real, std::enable_if_t<!std::is_integral_v<Real>, bool> = true>
+inline constexpr Real ldexp(Real arg, int exp)
+{
+    if(BOOST_MATH_IS_CONSTANT_EVALUATED(arg))
+    {
+        return boost::math::ccmath::abs(arg) == Real(0) ? arg :
+               boost::math::ccmath::isinf(arg) ? arg :
+               boost::math::ccmath::isnan(arg) ? arg :
+               exp == 0 ? arg :
+               arg * boost::math::ccmath::detail::intexp2<Real>(exp);
```
Comment (jzmaddock) [inline include/boost/math/ccmath/ldexp.hpp:91]: I think there's an issue with this approach: lets say arg is very large and exp is large and negative, in that situation the result of intexp2 can go to zero even though the result is expected to be finite.

I would simplify this and do something more like:

```
while(exp > 0)
{
   arg *=2;
   --exp;
}
while(exp < 0)
{
   arg /= 2;
   ++exp;
}
```

Which should return 0 or INF in the edge cases anyway?

**Comment by mborland** [comment]
This is clean on CI with the updated implementation.

**Pull request merged**